uber/site_sections/preregistration.py

```diff
@@ -191,12 +191,31 @@ def banned(self, id):
         attendee, group = self._get_unsaved(id)
         return {'attendee': attendee}
 
+    def freereg_payment(self, session):
+        charge = Charge(listify(self.unpaid_preregs.values()))
+        if charge.total_cost == 0:
+            for attendee in charge.attendees:
+                attendee.paid = c.HAS_PAID
+                attendee.amount_paid = attendee.total_cost
+                session.add(attendee)
+
+            for group in charge.groups:
+                group.amount_paid = group.default_cost - group.amount_extra
```

Comment:
As above, we don't want this behavior for free badges. You can remove this line and the entire for loop below.